pycozmo/activity.py

```diff
@@ -14,30 +16,389 @@
+
+
+class NeedsActivity(Activity):
+    def __init__(self,
+                 behavior_chooser: BehaviorChooser,
+                 *args, **kwargs):
+        self.behavior_chooser = behavior_chooser
+        super().__init__(*args, **kwargs)
+
+    def choose(self):
+        if self.behavior_chooser:
+            return self.behavior_chooser.choose()
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(
+            behavior_chooser=BehaviorChooser.from_json(data['behaviorChooser']),
+            activity_id=data['activityID'],
+            activity_type=data['activityType'],
+            strategy=data['activityStrategy']['type'])
+
+
+def from_dict(info: Dict) -> Activity:
+    if info['activityType'] == 'VoiceCommand':
+        return VoiceCommandActivity.from_json(info)
+
+    elif info['activityType'] == 'BehaviorsOnly':
+        return BehaviorsActivity.from_json(info)
+
+    elif info['activityType'] == 'Feeding':
+        return FeedingActivity.from_json(info)
+
+    elif info['activityType'] == 'Freeplay':
+        return FreeplayActivity.from_json(info)
+
+    elif info['activityType'] == 'Sparked':
+        return SparkedActivity.from_json(info)
+
+    elif info['activityType'] == 'BuildPyramid':
+        return PyramidActivity.from_json(info)
+
+    elif info['activityType'] == 'Socialize':
+        return SocializeActivity.from_json(info)
+
+    elif info['activityType'] == 'NeedsExpression':
+        return NeedsActivity.from_json(info)
+
+    else:
+        return Activity(
+            activity_id=info['activityID'],
+            activity_type=info['activityType'],
+            strategy=info['activityStrategy']['type']
+        )
+
+
+def get_activity_files(resource_dir: str) -> List[str]:
+    activity_files = [resource_dir + '/cozmo_resources/config/engine/behaviorSystem/activities_config.json']
+    activity_folder = resource_dir + '/cozmo_resources/config/engine/behaviorSystem/activities/'
+    for root, dirs, files in os.walk(activity_folder):
+        for name in files:
+            if name.endswith((".json")):
+                activity_files.append(os.path.join(root, name))
+    return activity_files
+
+
+def load_activities(resource_dir: str) -> Dict[str, Activity]:
+    """ Load activity map from cozmo resources. """
 
-def load_activities() -> Dict[str, Activity]:
-    # TODO: Load:
-    #  - cozmo_resources/config/engine/behaviorSystem/activities_config.json
-    #  - cozmo_resources/config/engine/behaviorSystem/activities/*/*.json
     activities = {}
+    activity_files = get_activity_files(resource_dir)
+
+    for filename in activity_files:
+        with open(filename, 'r') as cf:
+            filtered_json = ''
+            for line in cf.readlines():
+                if '//' in line:
+                    wordlist = line.split(' ')
+                    # get all words before '//'
+                    line = ' '.join(wordlist[:wordlist.index('//')])
```

Comment:
Uhh, they are using malformed JSON :/

Reply:
Yes, quite annoying haha I hope they are only using this format for the comments.